You-Can-Code-It/fndr
# Details page

# What this pull request does

## What

- Clicking over a company's name in the main page (list of company cards) leads the user to a Details Page
- The Details Page displays more infos about the company when compared to the company card (for example, the complete address)

## Why

- A Details Page allows the user to be more focused on and to obtain more information of a company
- Further features can be added in the Details Page, allowing the app to expand (e.g. a map displaying the location of the company with a pin on it, or a user's review section)

## How

- The `name` of a company in its card (main page, or companies' list) is a link to the Details Page: `/companies/:companyId`
- The Details Page is dynamically generated in the route `/companies/:companyId` (the "id" is already part of the props)
- The data from a specific company is fetched with `getServerSideProps` combined with `context.params.companyId`, and displayed in the Details Page

## Steppings to test this

- Go to the main page (list of companies, displayed in cards)
- Select one company, and click over its name
- You shall be forwarded to `/companies/:companyId`, and see the details of the company

## Anything else

- More info can be later added to the Details Page, so that the company card (in the main page) comprises only essential info, while the Details Page can grow more resources
